sdk/eventhub/azure-eventhubs/api_review.py

```diff
@@ -0,0 +1,230 @@
+
+class EventHubProducerClient(ClientBaseAsync):
+
+    eh_name  # type: str
+
+    def __init__(
+            self,
+            host: str,
+            event_hub_path: str,
+            credential: TokenCredential,
+            **kwargs
+    ) -> None:
+
+    async def __aenter__(self):
+
+    async def __aexit__(self, *args):
+
+    @classmethod
+    def from_connection_string(
+        cls, conn_str: str,
+        *,
+        event_hub_path: str = None,
+        logging_enable: bool = False,
+        http_proxy: dict = None,
+        auth_timeout: float = 60,
+        user_agent: str = None,
+        retry_total: int = 3,
+        transport_type=None) -> EventHubProducerClient:
+
+    async def send(
+        self,
+        event_data: Union[EventData, EventDataBatch, Iterable[EventData]],
+        *,
+        partition_key: Union[str, bytes] = None,
+        partition_id: str = None,
+        timeout: float = None) -> None:
+
+    async def create_batch(self, events: Iterable[EventData] = None, max_size: int = None) -> EventDataBatch:
```

Comment:
We are passing in optional EventData list to create batch to reduce the amount of code to write to start sending data.
I believe this is different to other languages.

Reply:
To clarify, this changes the user code from:
```python
batch = client.create_batch()
try:
    for event in my_event_list:
        batch.try_add(event)
except ValueError:
   # use a smaller batch
else:
    client.send_batch(batch)
```
Now becomes:
```python
try:
    batch = client.create_batch(my_event_list)
except ValueError:
    # use smaller batch
else:
    client.send_batch(batch)
```

Reply:
Could make this a non-breaking post-GA update.